core/src/main/java/org/elasticsearch/index/analysis/Analysis.java

```diff
@@ -179,11 +180,15 @@ public static CharArraySet parseCommonWords(Environment env, Settings settings,
     }
 
     public static CharArraySet parseArticles(Environment env, Settings settings) {
-        return parseWords(env, settings, "articles", null, null, settings.getAsBoolean("articles_case", false));
+        org.elasticsearch.Version version = settings.getAsVersion(IndexMetaData.SETTING_VERSION_CREATED, org.elasticsearch.Version.CURRENT);
+        boolean articlesCase = settings.getAsBooleanLenientForPreEs6Indices(version, "articles_case", false);
+        return parseWords(env, settings, "articles", null, null, articlesCase);
     }
 
     public static CharArraySet parseStopWords(Environment env, Settings settings, CharArraySet defaultStopWords) {
-        return parseStopWords(env, settings, defaultStopWords, settings.getAsBoolean("stopwords_case", false));
+        org.elasticsearch.Version version = settings.getAsVersion(IndexMetaData.SETTING_VERSION_CREATED, org.elasticsearch.Version.CURRENT);
```

Comment:
Just to be clear: Do you mean:

(A) The method `Settings#getAsVersion()` should not even have a parameter for the default value.
(B) The call-sites I have added should not allow a default value.